oeps/oep-38-Data-Modeling.rst

```diff
@@ -0,0 +1,442 @@
+
+| OEP | :doc:` |
+
++---------------+------------------------------------------------------------+
+
+| Title | Data Modeling Best Practices |
+
++---------------+------------------------------------------------------------+
+
+| Last Modified | 2019-08-01 |
+
++---------------+------------------------------------------------------------+
+
+| Author | Brian Beggs <bbeggs@edx.org> |
+
++---------------+------------------------------------------------------------+
+
+| Arbiter | Albemarle |
+
++---------------+------------------------------------------------------------+
+
+| Status | Provisional |
+
++---------------+------------------------------------------------------------+
+
+| Type | Architecture |
+
++---------------+------------------------------------------------------------+
+
+| Created | 2019-05-01 |
+
++---------------+------------------------------------------------------------+
+
+| Review Period | 2019-08-01 - 2019/08/16 |
+
++---------------+------------------------------------------------------------+
+
+| Resolution | - 'Original pull request' |
+
+| | - 'Update pull request' |
+
++---------------+------------------------------------------------------------+
+
+| References | - 'OEP-002' |
+
++---------------+------------------------------------------------------------+
+
+
+.. \_Original pull request:
+
+.. \_Update pull request:
+
+.. \_OEP-002:
+https://open-edx-proposals.readthedocs.io/en/latest/oep-0002-bp-repo-metadata.html
+
+Abstract
+
+========
+
+In order to obtain the highest possible value from data collected in the
+edX ecosystem, this document attempts to provide guiding thoughts and
+principles on data modeling.
+
+These data modeling guidelines are broken down into three sections:
+
+-  Guiding Principles
+
+-  Data Modeling Standards
+
+-  Best Practices
+
+Guiding Principles - This section is intended to give the reader a
+framework to think about data modeling at edX. While no specific
+instructions are given in this section, it should act as a framework to
+help the reader think about data modeling on the whole at edX.
+
+Data modeling standards - The core guidelines that everyone must adhere
+to when creating new data models at edX. These standards are designed to
+ensure edX can gain the highest value and insights from the data. They
+should be applied as the basic standard for all applications in the
+OpenedX ecosystem.
+
+Data Best Practices - These practices are designed to help teams create
+rich and efficient data models within the edX ecosystem. They are not
+standards but guidelines to help teams think about how they are storing
+data.
+
+This OEP explicitly does not cover:
+
+-  Data warehousing
+
+-  Data pipelines
+
+-  Event data
+
+Motivation
+
+==========
+
+The goal of this OEP is to provide a framework about how to think about
+data modeling at edX and provide guidelines on what should and should
+not be done when creating data models.
```

Comment:
It seems like the purpose of the motivation section might be to provide some context and background that clarifies the events that culminated to the authoring of this document, or a description of scenarios that it is indented to prevent. Otherwise, it just feels like a reiteration of the abstract.

Reply:
I have removed this section.